msprime/demography.py

```diff
@@ -2431,6 +2459,301 @@ def _american_admixture_model():
         demography.add_population_split(5920, derived=["AMH"], ancestral="ANC")
         return demography
 
+    def to_demes(self) -> demes.Graph:
+        """
+        Creates a :class:`demes.Graph` object from the demography.
+
+        The returned graph can be saved as a
+        `Demes-format <https://popsim-consortium.github.io/demes-spec-docs/main/>`_
+        YAML file using the :ref:`demes <demes:sec_introduction>` API.
+
+        .. code::
+
+            import demes
+
+            demography = msprime.Demography.island_model([1000] * 3, 1e-5)
+            graph = demography.to_demes()
+            demes.dump(graph, "island_model.yaml")
+
+        Or plotted using the `demesdraw <https://github.com/grahamgower/demesdraw>`_
+        visualisation package.
+
+        .. code::
+
+            import demesdraw
+
+            demography = msprime.Demography.island_model([1000] * 3, 1e-5)
+            graph = demography.to_demes()
+            ax = demesdraw.tubes(graph)
+            ax.figure.savefig("island_model.pdf")
+
+        :return: A :class:`demes.Graph` object corresponding to the demography.
+        :rtype: demes.Graph
+        """
+        resolved = self.validate()
+        b = demes.Builder()
+        for pop in resolved.populations:
+            initial_epoch = dict(
+                end_size=pop.initial_size, growth_rate=pop.growth_rate, end_time=0
+            )
+            b.add_deme(
+                pop.name,
+                description=pop.description if pop.description else None,
```

Comment:
Not sure the ``if`` is doing much here? Might as well pass through the empty string value

(I'm not sure why I made the default the empty string rather than None, maybe I did this when demes required a ``description`` field? I think it would probably be better to have the description = None if not specified in Demography.add_population()``

Reply:
Demes doesn't allow empty strings (but allows None).